### ISSUE TITLE
Migrate CircleCI to a 2.0 configuration

### ISSUE THREAD

**Issue by Arcanemagus** [AtomLinter/atom-linter-perlcritic#49]
CircleCI will be [removing support for 1.0](https://circleci.com/blog/sunsetting-1-0/) configurations as of 2018-08-31. This is one of the repositories within this organization that is currently still configured to use a 1.0 configuration and needs to be migrated to the 2.0 format.

An automated translation can be found [here](https://circleci.com/api/v1.1/project/github/AtomLinter/atom-linter-perlcritic/config-translation), however it is likely that a better method would be to follow the Workflow based example from [AtomLinter/atom-minimap-linter](https://github.com/AtomLinter/atom-minimap-linter/blob/master/.circleci/config.yml).

**Comment by whity** [comment]
Hi @Arcanemagus do you mind to take a look at the 'dev' branch. I already added the circle ci new configuration, but something is not right, since I get some warnings (I don't get them locally):

`FFFFFFFFFF

The perlcritic provider for Linter
  has settings that control the output
    it allows controlling whether the severity shows
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:50:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it allows controlling whether the explanation shows
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:66:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it allows controlling whether the policy shows
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:82:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
  allows controlling the forced severity shown
    it defaults to not forcing the policy
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:105:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it supports forcing gentle severity
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:112:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it supports forcing stern severity
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:119:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it supports forcing harsh severity
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:126:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
    it supports forcing cruel severity
      TypeError: Cannot read property 'length' of null
        TypeError: Cannot read property 'length' of null
        at /tmp/project/spec/atom-linter-perlcritic-spec.js:133:22
        at Generator.next (<anonymous>)
        at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
        at <anonymous>
        at process._tickCallback (internal/process/next_tick.js:109:7)
  it shows nothing wrong with a valid file
    TypeError: Cannot read property 'length' of null
      TypeError: Cannot read property 'length' of null
      at /tmp/project/spec/atom-linter-perlcritic-spec.js:33:20
      at Generator.next (<anonymous>)
      at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:109:7)
  it properly shows messages for a problematic file
    TypeError: Cannot read property 'length' of null
      TypeError: Cannot read property 'length' of null
      at /tmp/project/spec/atom-linter-perlcritic-spec.js:40:20
      at Generator.next (<anonymous>)
      at step (/tmp/project/spec/atom-linter-perlcritic-spec.js:3:273)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:109:7)


Finished in 1.114 seconds
10 tests, 10 assertions, 10 failures, 0 skipped` 

Since you're most used to atom (I don't use it :p), maybe you easily find what's happening.
If you don't have the time, thanks anyway, I'll continue with this another day.

Thanks.

**Comment by Arcanemagus** [comment]
Sure, I'll take a look 👍.

**Comment by Arcanemagus** [comment]
Fixed the way it was getting Perl installed (`apt-get update` before install) and fixed the way `apm` was getting called to work on both builds.

Unfortunately... it looks like something to do with the CircleCI _host_ is breaking the installation of dependencies of the `Perl::Critic` module, preventing it from getting installed. I've ran the exact same steps on the same image on my local machine and it works fine. I'm not sure what the solution to this is yet...
